c2rust-analyze/src/c_void_casts.rs

```diff
@@ -251,6 +256,27 @@ pub struct CVoidCasts<'tcx> {
     to: CVoidCastsUniDirectional<'tcx>,
 }
 
+pub fn rv_place<'tcx>(rv: &Rvalue<'tcx>) -> Option<Place<'tcx>> {
+    use Rvalue::*;
+    match rv {
+        Use(op) => op.place(),
+        Repeat(op, _) => op.place(),
+        Ref(_, _, p) => Some(*p),
```

Comment:
What's the purpose of this function?  It doesn't seem like a very meaningful operation, since the `Place`s it returns are used in all sorts of different ways depending on what kind of `Rvalue` was passed.  For example, some `Rvalue`s read from `rv_place(rv)`, while others don't.

Looking at the use site below, it seems like this is part of the logic for propagating through multiple levels of casts, as in `p as *const c_char as *const c_void`.  If that's the case, can this be reduced to the `Cast` case only, where the meaning is clear?